bookshelf/src/DataModel/Datastore.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Samples\Bookshelf\DataModel;
+
+use Google_Service_Datastore;
+
+/**
+ * Class DatastoreModel implements the DataModel with a Google Data Store.
+ *
+ * Incomplete and untested.
+ */
+class Datastore implements DataModelInterface
+{
+    protected $columns = [
+        'id'            => 'integer',
+        'title'         => 'string',
+        'author'        => 'string',
+        'publishedDate' => 'dateTime',
+        'imageUrl'      => 'string',
+        'description'   => 'string',
+        'createdBy'     => 'string',
+        'createdById'   => 'string',
+    ];
+
+    public function __construct($datastoreDatasetId)
+    {
+        $this->datasetId = $datastoreDatasetId;
+        $client = new \Google_Client();
+        $client->setScopes([
+            Google_Service_Datastore::CLOUD_PLATFORM,
+            Google_Service_Datastore::DATASTORE,
+            Google_Service_Datastore::USERINFO_EMAIL,
```

Comment:
No, it throws an `invalid_scope` scope or a `401` when I remove all/some of them (respectively)